x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/import_rules_route.ts

```diff
@@ -18,17 +18,19 @@ import { getIndexExists } from '../../index/get_index_exists';
 import {
   callWithRequestFactory,
   getIndex,
-  createImportErrorObject,
-  transformImportError,
-  ImportSuccessError,
+  createBulkErrorObject,
+  ImportRuleResponses,
 } from '../utils';
 import { createRulesStreamFromNdJson } from '../../rules/create_rules_stream_from_ndjson';
 import { ImportRuleAlertRest } from '../../types';
-import { transformOrImportError } from './utils';
 import { updateRules } from '../../rules/update_rules';
 import { importRulesQuerySchema, importRulesPayloadSchema } from '../schemas/import_rules_schema';
 import { KibanaRequest } from '../../../../../../../../../src/core/server';
 
+type PromiseFromStreams = ImportRuleAlertRest | Error;
+
+const CHUNK_PARSED_OBJECT_SIZE = 10;
```

Comment:
Maybe comment the reason for adding chunk size of 10 because of default max listeners for event emitters was throwing a warning for possible memory leak?